code/src/components/LoadingIndicator.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import Loader from 'react-loader-spinner'
+
+export const LoadingIndicator = () => {
+    const isLoading = useSelector((state) => state.ui.isLoading)
+
+    return (
+        <>
+            {isLoading && (
+                <Loader
+                    type="ThreeDots"
```

Comment:
and also here, indentation. I'll stop commenting about this now but take a look in your vscode settings.